aiohttp/http.py

```diff
@@ -27,7 +28,7 @@
 
     # .http_websocket
     'WS_CLOSED_MESSAGE', 'WS_CLOSING_MESSAGE', 'WS_KEY',
-    'WebSocketReader', 'WebSocketWriter', 'do_handshake',
+    'WebSocketReader', 'WebSocketWriter', 'ws_ext_gen', 'ws_ext_parse',
```

Comment:
Why expose `ws_ext_gen` and `ws_ext_parse` then?

Reply:
It's a part of internal API, why not.

My main intention for `do_handshake` changes is not preventing exposing new functions but not converting `http` exceptions into `web` ones.
PR raises proper exceptions directly. I believe it simplifies understanding a little, plus stack traces are shorter.

Reply:
Aha, I see. Thanks!